janrain/jump.ios
# Fixes Compiler Errors with iOS 10 SDK in Xcode 8 (Installed via CocoaPods)

Fixes #38

## What it Does

Currently, when installing via Cocoapods, the project fails to build with errors due to identical folder reference names. The folder names in question are those nested within the `Janrain/JREngage/Resources/images` and `Janrain/JREngage/Resources/images-2x` folders. See #38 for more information. This pull request changes the name of all subfolders within `images-2x` by appending `-2x` to them as well. Screenshots to compare before / after:

![screen shot 2017-06-26 at 2 22 14 pm](https://user-images.githubusercontent.com/7883805/27554164-7ec650ec-5a7b-11e7-8349-99e0be16c02b.png)
![screen shot 2017-06-26 at 2 21 17 pm](https://user-images.githubusercontent.com/7883805/27554171-884177be-5a7b-11e7-99e0-4e3e5a6c5a50.png)

Additionally, once the duplicate resources issue is resolved, attempting to `import Janrain` results in compiler errors:

* `ROpenIDAppAuthGoogleDelegate.h:28:9: 'AppAuth.h' file not found`
* `JRProvidersController.h:39:53: Cannot find protocol declaration for 'JRUserInterfaceDelegate'`

fb41c84 fixes these errors.

## How to Test

To test, add the following line to your Podfile in a project without Janrain installed, and `pod install`:

```
pod 'Janrain', :git => 'https://github.com/Lickability/jump.ios.git', :branch => 'remove-duplicate-resource-references'
```

Once installing has completed import Janrain into any of our project’s source files (e.g. `import Janrain` in Swift), build your Xcode workspace, and see that there are no errors.
